src/js/components/Spinner/Spinner.js

```diff
@@ -0,0 +1,102 @@
+import React, {
+  isValidElement,
+  forwardRef,
+  useContext,
+  useEffect,
+} from 'react';
+import { ThemeContext } from 'styled-components';
+
+import { AnnounceContext } from '../../contexts/AnnounceContext';
+import { Box } from '../Box';
+import { defaultProps } from '../../default-props';
+
+const BasicSpinner = ({ ref, size, ...rest }) => (
+  <Box height={size} width={size} ref={ref} {...rest} />
+);
+/**
+ * If the user is calling <Spinner>…</Spinner> with children, it will take
+ * precedence over theme styling. Yet, it will still inherit the
+ * default animation and size of the spinner, and of course any additional
+ * given props.
+ *
+ * If the user is providing an icon/svg via the theme.spinner.icon,
+ * the Spinner will use it as a child and will include all its relevant
+ * theme props (size/color/pad…) as well,
+ * user will only need to type <Spinner />.
+ * If the icon has its own animation, user can turn it off via the theme.
+ *
+ * If none of the above is provider, <Spinner /> will provide its default
+ * border, size and friends, all configurable via theme or props.
+ */
+const Spinner = forwardRef(
+  ({ children, color: colorProp, size, message, ...rest }, ref) => {
+    const theme = useContext(ThemeContext) || defaultProps.theme;
+    const announce = useContext(AnnounceContext);
+
+    useEffect(() => {
+      if (message?.start) {
```

Comment:
I think we should allow `message` to be a string and to announce only on mount. I think there will be plenty of uses for this case, as the completion could be announced by the component/content rendered.